demos/ready.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// eslint-disable no-unused-vars no-var
+
+/**
+ * Adds the given event handler to the queue. It will be executed asynchronously after all external JS and CSS resources
+ * have finished loading (as determined by continuous long-polling with a timeout). If this function is called after all
+ * resources have finished loading, the given handler function will be invoked synchronously (in the same call stack).
+ * Handlers are invoked in FIFO order.
+ * @param {function() : undefined} handler
+ */
+window.demoReady = (function() {
+  var TIMEOUT_MS = 60 * 1000;
+  var DELAY_MS = 100;
+
+  var isReadyCached = false;
+  var handlers = [];
+  var testDom = null;
+  var startTimeMs = null;
+  var timer = null;
+
+  function isReady() {
+    if (isReadyCached) {
+      return true;
+    }
+    ensureDetectionDom();
+    isReadyCached = Boolean(window.mdc) && getComputedStyle(testDom).position === 'relative';
+    return isReadyCached;
+  }
+
+  function ensureDetectionDom() {
+    if (testDom) {
+      return;
+    }
+    testDom = document.createElement('div');
+    testDom.classList.add('demo-ready-detect');
+    document.body.appendChild(testDom);
+  }
+
+  function removeDetectionDom() {
+    if (!testDom) {
+      return;
+    }
+    document.body.removeChild(testDom);
+    testDom = null;
+  }
+
+  function startTimer() {
+    if (isReady() || timer) {
```

Comment:
If I'm following the logic correctly...I think this conditional can be removed since
- it seems like `timer` will always be null here
- in the case where `isReady()` is true but `timer` is null, the handler will never get called

Reply:
Good catch! Removed the `isReady()` check.

`timer` will only be `null` the first time `addHandler()` is called, so I think we need to keep that check.